GVFS/GVFS.Service/GVFSMountProcess.Mac.cs

```diff
@@ -1,52 +1,72 @@
 ﻿using GVFS.Common;
-using GVFS.Common.Tracing;
-using GVFS.Service.Handlers;
-using System;
+using GVFS.Common.Tracing;
+using System.Diagnostics;
+using System.IO;
 
 namespace GVFS.Service
 {
-    public class GVFSMountProcess : IDisposable
+    public class GVFSMountProcess : IRepoMounter
     {
-        private const string ParamPrefix = "--";
+        private const string ExecutablePath = "/bin/launchctl";
 
-        private readonly ITracer tracer;
+        private MountLauncher processLauncher;
+        private GVFSPlatform platform;
 
-        public GVFSMountProcess(ITracer tracer, int sessionId)
+        public GVFSMountProcess(MountLauncher processLauncher = null, GVFSPlatform platform = null)
         {
-            this.tracer = tracer;
+            this.processLauncher = processLauncher ?? new MountLauncher();
+            this.platform = platform ?? GVFSPlatform.Instance;
         }
 
-        public bool Mount(string repoRoot)
+        public bool MountRepository(string repoRoot, int sessionId, ITracer tracer)

```

Comment:
I'm not a fan of inserting the tracer in the method. The tracer is something that exists all the time and is easier to inject at construction time. It also makes this change smaller, by keeping the tracer as an object member. 
Same goes for `LaunchProcess`. This was something that I thought was particularly valuable when designing the object model for `GitMaintenanceStep` and its extensions: the `GitMaintenanceStep` constructor holds the `GVFSContext` as a class member so we don't need to pass environment parameters in all of the utility methods.

This seems like something we should start standardizing more, but requires buy-in from the group. Thoughts, @wilbaker, @jrbriggs, @kewillford, @jamill, @jeschu1?

I understand the value in changing the location of `sessionId`, especially because it was ignored before.

Reply:
> This seems like something we should start standardizing more, 

I like this. I think we generally include tracer in the object construction where possible elsewhere.

Reply:
I know that we talked about this a little in Jameson's meeting and dependencies for the class should be passed to the constructor like you have suggested.  I'm also for minimalizing dependencies so that if the class only needs the `ITracer` then only pass that in and not the whole `context` object but if the class needs multiple things that are in the `context` then pass the `context` instead of multiple objects.

Reply:
Moved it to the Constructor.